### PR TITLE
Create crunch.lang

### PR DESCRIPTION
**Pull request by 1011X**
Crunch syntax highlighting for gedit

### CODE REVIEW

File: syntaxes/crunch.lang [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    </context>
+    
+    <!--
+    <context id="reserved" style-ref="def:reserved">
+    </context>
+    -->
+
+    <context id="types" style-ref="type">
+      <match extended="true">
+        \b(bool|unit|int|float|str)\b
```
Comment (Kixiron) [inline syntaxes/crunch.lang:58]: I don't know if it's possible, but also allowing `&` and `&mut` somehow would be great

File: syntaxes/crunch.lang [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      <keyword>let</keyword>
+      <keyword>loop</keyword>
+      <keyword>match</keyword>
+      <keyword>pkg</keyword>
+      <keyword>return</keyword>
+      <keyword>then</keyword>
+      <keyword>trait</keyword>
+      <keyword>type</keyword>
+      <keyword>where</keyword>
+      <keyword>while</keyword>
```
[the remark this answers is not in the document]
Comment (1011X) [reply syntaxes/crunch.lang:48]: should i add `do` too? the sample code in the readme has it and seems like a keyword

File: syntaxes/crunch.lang [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    <context id="line-comment" style-ref="comment" end-at-line-end="true" class-disabled="no-spell-check" class="comment">
+      <start>::</start>
+      <include>
+        <context ref="def:in-comment"/>
+      </include>
+    </context>
+
+    <context id="line-doc-comment" style-ref="def:doc-comment" end-at-line-end="true" class="comment" class-disabled="no-spell-check">
+      <start>:::</start>
+    </context>
```
[the remark this answers is not in the document]
Comment (1011X) [reply syntaxes/crunch.lang:162]: oof, that would be, uh, really difficult.

File: syntaxes/crunch.lang [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    <context id="byte-string" style-ref="string" class="string" class-disabled="no-spell-check">
+      <start>b"</start>
+      <end>"</end>
+      <include>
+        <context ref="byte-string-escape"/>
+        <context ref="def:line-continue"/>
+      </include>
+    </context>
```
[the remark this answers is not in the document]
Comment (1011X) [reply syntaxes/crunch.lang:139]: hmm, idk how to do arbitrary expressions, but i can do rust's format specifiers. e.g. `"hello {:0b}"`. is that good?

File: syntaxes/crunch.lang (hunk 1 above)
Comment (1011X) [reply syntaxes/crunch.lang:58]: hmm, i think & is distinguishable enough to not need it. the `mut` part *will* get highlighted tho.

[the remark this answers is not in the document]
Comment (1011X) [reply syntaxes/crunch.lang:58]: yep!

File: syntaxes/crunch.lang (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kixiron) [reply syntaxes/crunch.lang:162]: That's alright, I didn't know if gedit allowed language integrations like textmate does

File: syntaxes/crunch.lang (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kixiron) [reply syntaxes/crunch.lang:48]: Yes, both `do` is a keyword, and I also forgot about `async`

**Comment by Kixiron** [comment]
I've been trying this on Ubuntu 20.04, gedit v3.36.1, but I get the following error ```
(gedit:35026): GtkSourceView-WARNING **: 00:03:26.999: Failed to load '/usr/share/gtksourceview-4/language-specs/crunch.lang': invalid context reference “crunch:reserved”
```

File: syntaxes/crunch.lang (hunk 3 above)
[the remark this answers is not in the document]
Comment (1011X) [reply syntaxes/crunch.lang:162]: oh it does, i just don't know how to get that going and it's getting late here lol

**Comment by 1011X** [comment]
That's been fixed in the new change.

List of possible future changes:
+ add formatting parameters like in rust
+ display doc-comments as markdown, if possible
+ make annotations more general

**Comment by Kixiron** [comment]
Thank you so much, it looks fantastic!

**Pull request merged**